spec/stack.js

```diff
@@ -11,12 +11,48 @@ describe('Stack', () => {
     expect(Stack).to.be.a('function')
   })
 
+  context('length()', () => {
+    it('returns the size of the stack', () => {
+      const myStack = new Stack()
+      myStack.push('foo')
+      myStack.push('bar')
+      expect(myStack.length()).to.eql(2)
+    })
+  })
+
+  context('isEmpty()', () => {
+    it('returns true if the stack is empty', () => {
+      const myStack = new Stack()
+      myStack.push('foo')
+      myStack.push('bar')
+      expect(myStack.isEmpty()).to.eql(false)
+    })
+  })
+
   context('push()', () => {
     it('pushes an element to the top of the stack.', () => {
       const myStack = new Stack()
+      myStack.push('foo')
+      expect(myStack.length()).to.deep.eql(1)
+    })
+  })
+
+  context('pop()', () => {
+    it('pops the top element of the stack', () => {
+      const myStack = new Stack()
+      myStack.push('foo')
+      myStack.push('bar')
+      expect(myStack.pop()).to.eql(['bar'])
+    })
+  })
 
-      expect(() => myStack.push('foo'))
-        .to.alter(() => myStack.length(), { from: 0, to: 1 })
+  context('peek()', () => {
+    it('peeks the top element of the stack', () => {
```

Comment:
I might include a test to ensure that the size of the stack *doesn't* change with `peek` (but more because I've seen many students mess this up, than for any other reason 😄 ).